lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -1,122 +1,125 @@
-    weapon_frag       = 0,
-    weapon_rpg        = 0,
-    weapon_fists      = 0,
-    m9k_fists         = 0,
-    m9k_m98b          = 2,
-    ins2_atow_rpg7    = 3,
-    m9k_matador       = 3,
-    m9k_m202          = 3,
-    m9k_rpg7          = 3,
-    m9k_minigun       = 4,
+CFCPvpMovespeed = {
+    -- default run and walk speed with 0 weapons
+    baseRunSpeed = 400,
+    baseWalkSpeed = 200,
+
+    -- minimum run and walk speed ( must be greater than 0 )
+    minRunSpeed = 70,
+    minWalkSpeed = 35,
+
+    -- Weapon weights, weapons not in the table have a weight of 1
+    weaponWeights = {
+        weapon_physgun    = 0,
+        weapon_physcannon = 0,
+        none              = 0,
+        laserpointer      = 0,
+        remotecontroller  = 0,
+        gmod_tool         = 0,
+        gmod_camera       = 0,
+        weapon_357        = 0,
+        weapon_ar2        = 0,
+        weapon_crossbow   = 0,
+        weapon_crowbar    = 0,
+        weapon_pistol     = 0,
+        weapon_shotgun    = 0,
+        weapon_smg1       = 0,
+        weapon_medkit     = 0,
+        weapon_frag       = 0,
+        weapon_rpg        = 0,
+        weapon_fists      = 0,
+        m9k_fists         = 0,
+        m9k_m98b          = 2,
+        ins2_atow_rpg7    = 3,
+        m9k_matador       = 3,
+        m9k_m202          = 3,
+        m9k_rpg7          = 3,
+        m9k_minigun       = 4,
+    },
+
+    cfcHookPrefix = "CFC_PlyMS_"
 }
-pvpMoveSpeed = {}
 
 -- Helper Functions --
-local cfcHookPrefix = "CFC_PlyMS_"
-local function generateCFCHook( hookname )
-    return cfcHookPrefix .. hookname
+function CFCPvpMovespeed:generateCFCHook( hookname )
+    return self.cfcHookPrefix .. hookname
 end
 
-local function getPlayerPvpMode( ply )
+function CFCPvpMovespeed.playerIsInBuild( ply )
     return ply:GetNWBool( "CFC_PvP_Mode", false )
 end
 
-local function playerIsInBuild( ply )
-    return not getPlayerPvpMode( ply )
-end
-
-local function isValidPlayer( ply )
+function CFCPvpMovespeed.isValidPlayer( ply )
     return IsValid( ply ) and ply:IsPlayer()
 end
 
-local function movementMultiplier( totalWeight )
+function CFCPvpMovespeed.movementMultiplier( totalWeight )
     if totalWeight < 1 then return 1 end
     local multiplier = 1 - ( 1.9 ^ totalWeight  ) / 100
     return math.Clamp( multiplier, 0, 1 )
 end
 
-local function setSpeedFromWeight( ply, totalWeight )
+function CFCPvpMovespeed:setSpeedFromWeight( ply, totalWeight )
     local multiplier = movementMultiplier( totalWeight )
 
-    local newRunSpeed = baseRunSpeed * multiplier
-    local newWalkSpeed = baseWalkSpeed * multiplier
+    local newRunSpeed = self.baseRunSpeed * multiplier
+    local newWalkSpeed = self.baseWalkSpeed * multiplier
+
     ply:SetRunSpeed( math.max( newRunSpeed, minRunSpeed ) )
     ply:SetWalkSpeed( math.max( newWalkSpeed, minWalkSpeed ) )
 
     if newWalkSpeed < 100 then
         ply:ChatPrint( "You are holding too many weapons! Drop some to regain speed." )
```

Comment:
```suggestion
        ply:ChatPrint( "You are holding too many weapons! /Drop some to regain speed." )
```
Add an invisible /drop tutorial here